types/benchmark_test.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"math/rand"
+	"testing"
+)
+
+func BenchmarkDefaultTypeForValue(b *testing.B) {
+	b.Run("RandUint64__overhead cost   ", func(b *testing.B) {
```

Comment:
This is irrelevant, no need to benchmark rand.Uint64.
When we do benchmark, we should bench the real function it self.
This is an example, you may want to take a look.

```go
func BenchmarkFastLenUint(b *testing.B) {
	nums := make([]uint64, 10)
	for i := range nums {
		nums[i] = rand.Uint64() %1000
	}

	b.Run("A", func(b *testing.B) {
		b.StartTimer()
		for i:=0; i<b.N; i++{
			LenUint64A(nums[int(i)%len(nums)])
		}
	})

	b.Run("C", func(b *testing.B) {
		b.StartTimer()
		for i:=0; i<b.N; i++{
			LenUint64C(nums[int(i)%len(nums)])
		}
	})


	b.Run("B", func(b *testing.B) {
		b.StartTimer()
		for i:=0; i<b.N; i++{
			LenUint64B(nums[int(i)%len(nums)])
		}
	})
}
```